src/api/emulations/nfs.js

```diff
@@ -181,11 +196,88 @@ class NFS {
   * @returns {Promise<File>} - the same file
   **/
   update(fileName, file, version) {
-    return lib.file_update(this.mData.app.connection, this.mData.ref, fileName,
+    return lib.dir_update_file(this.mData.app.connection, this.mData.ref, fileName,
                            file.ref.ref(), version)
       .then(() => { file.version = version; })  // eslint-disable-line no-param-reassign
       .then(() => file);
   }
+
+  /**
+  * Delete a file from path. Directly commit to the network.
+  * @param {(String|Buffer)} fileName
+  * @param {Number} version
+  * @returns {Promise}
+  **/
+  delete(fileName, version) {
+    return lib.dir_delete_file(this.mData.app.connection, this.mData.ref, fileName, version);
+  }
+
+  /**
+  * Open a file for reading or writing.
+  *
+  * OPEN MODES:
+  *  /// Replaces the entire content of the file when writing data.
+  *  const OPEN_MODE_OVERWRITE = 1;
+  *  /// Appends to existing data in the file.
+  *  const OPEN_MODE_APPEND = 2;
+  *  /// Open file to read.
+  *  const OPEN_MODE_READ = 4;
+  *  /// Read entire contents of a file.
+  *  const FILE_READ_TO_END = 0;
+  *
+  * These constants are declared in ../../consts.js and imported in this module
+  *
+  * @param {File} file
+  * @param {Number} openMode
+  * @returns {Promise<FileContextHandle>}
+  **/
+  open(file, openMode) {
+    const fileRef = file.ref ? file.ref.ref() : new File(file).ref.ref();
+    return lib.file_open(this.mData.app.connection, fileRef, openMode);
+  }
+
+  /**
+  * Get file size
+  * @param {FileContextHandle} fileContextHandle
+  * @returns {Promise<Number>}
+  **/
+  size(fileContextHandle) {
```

Comment:
I think this function should be a method of the `File` class which internally keeps its `fileContextHandle`. We don't expose handles from the nodejs API but only JS objects.